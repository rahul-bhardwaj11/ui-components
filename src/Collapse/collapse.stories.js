import React from 'react';

import { storiesOf } from '@storybook/react';
import Collapse from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, object } from '@storybook/addon-knobs';

const stories = storiesOf('Collapse', module);
stories.addDecorator(withKnobs);

const customPanelStyle = {
  borderRadius: 8,
  marginBottom: 24,
  border: 0,
  overflow: 'hidden'
};

const options = [
  { header: 'This is panel header 1', content: ['First Item'] },
  { header: 'This is panel header 2', content: '<b>Second Item</b>' },
  { header: 'This is panel header 3', content: 'Third Item' }
];

stories.add(
  'Default Collapse',
  withInfo('Basic usage of the Collapse')(() => (
    <Collapse options={object('options', options)} />
  ))
);
stories.add(
  'Custom Collapse',
  withInfo('Basic usage of the Custom Collapse')(() => (
    <Collapse
      options={object('options', options)}
      panelStyle={object('panelStyle', customPanelStyle)}
    />
  ))
);
