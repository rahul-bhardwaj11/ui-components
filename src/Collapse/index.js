import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AntCollapse from 'antd/lib/collapse';
import 'antd/lib/collapse/style/index.css';
import StringToHTML from '../StringToHTML';
import styled from 'styled-components';
import theme from '../styles/theme';

const MtCollapse = styled.div`
  .ant-collapse {
    font-family: inherit;
    border: 0px;
    border-radius: 0px;
    background: transparent;
  }
  .ant-collapse {
    & > .ant-collapse-item {
      &:last-child {
        border-radius: 0px;
      }
    }
    & > .ant-collapse-item {
      &:last-child {
        & > .ant-collapse-header {
          border-radius: 0px;
        }
      }
    }
  }
  .ant-collapse-header {
    font-weight: bold;
    background: ${theme.colors.WHITE};
  }
  .ant-collapse {
    & > .ant-collapse-item {
      & > .ant-collapse-header {
        .arrow {
          font-size: 0px;
          width: 0;
          height: 0;
          border-left: 6px solid transparent;
          border-right: 6px solid transparent;
          border-top: 7px solid ${theme.colors.OUTER_SPACE};
          border-radius: 2px;
          margin-bottom: 20px;
          position: absolute;
          left: 91%;
          transform: rotate(-90deg);
        }
      }
    }
  }
  .ant-collapse {
    & > .ant-collapse-item {
      & > .ant-collapse-header[aria-expanded='true'] {
        .arrow {
          transform: rotate(0deg);
          position: absolute;
          left: 91%;
        }
      }
    }
  }

  .ant-collapse-content {
    border-top: 1px;
  }
`;

class Collapse extends Component {
  static propTypes = {
    options: PropTypes.any.isRequired,
    accordion: PropTypes.bool,
    panelStyle: PropTypes.object
  };

  render() {
    let { options, panelStyle } = this.props;
    return (
      <MtCollapse>
        <AntCollapse defaultActiveKey={['0']} {...this.props}>
          {options.map((option, index) => {
            return (
              <AntCollapse.Panel
                key={option.value || index}
                header={option.header}
                style={panelStyle}
                className={option.className}
                forceRender={option.forceRender}
              >
                {typeof option.content === 'string' ? (
                  <StringToHTML content={option.content} />
                ) : (
                  option.content
                )}
              </AntCollapse.Panel>
            );
          })}
        </AntCollapse>
      </MtCollapse>
    );
  }
}
Collapse.Panel = AntCollapse.Panel;
export default Collapse;
