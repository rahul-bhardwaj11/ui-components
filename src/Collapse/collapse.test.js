import React from 'react';
import renderer from 'react-test-renderer';
import Collapse from './index';

describe('Collapse', () => {
  it('matches  default snapshot', () => {
    const options = [
      { header: 'This is panel header 1', content: ['First Item'] },
      { header: 'This is panel header 2', content: '<b>Second Item</b>' },
      { header: 'This is panel header 3', content: 'Third Item' }
    ];
    const tree = renderer.create(<Collapse options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
