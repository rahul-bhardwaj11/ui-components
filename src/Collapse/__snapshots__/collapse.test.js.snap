// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse matches  default snapshot 1`] = `
<div
  className="sc-bdVaJa UsBxu"
>
  <div
    className="ant-collapse"
    role={null}
    style={undefined}
  >
    <div
      className="ant-collapse-item ant-collapse-item-active"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded="true"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          className="arrow"
        />
        This is panel header 1
      </div>
      <div
        className="ant-collapse-content ant-collapse-content-active"
        role={null}
      >
        <div
          className="ant-collapse-content-box"
        >
          First Item
        </div>
      </div>
    </div>
    <div
      className="ant-collapse-item"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded="false"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          className="arrow"
        />
        This is panel header 2
      </div>
    </div>
    <div
      className="ant-collapse-item"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded="false"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          className="arrow"
        />
        This is panel header 3
      </div>
    </div>
  </div>
</div>
`;
